packages/api/src/webhooks/cannon.test.ts

```diff
@@ -0,0 +1,228 @@
+      url: "http://localhost:30000/webhook",
+      // url: 'https://livepeer.com/'
+    };
+
+    ({
+      client,
+      adminUser,
+      adminToken,
+      nonAdminUser,
+      nonAdminToken,
+    } = await setupUsers(server));
+    console.log("beforeALL done");
+  });
+
+  beforeAll(async () => {
+    webhookServer = express();
+    return new Promise<void>((resolve, reject) => {
+      listener = webhookServer.listen(30000, function (err) {
+        if (err) {
+          return reject(err);
+        }
+        const port = listener.address().port;
+        testHost = `http://localhost:${port}`;
+        console.log("Example app listening at http://%s", testHost);
+        resolve();
+      });
+    });
+  });
+
+  afterAll(async () => {
+    await new Promise<void>((resolve, reject) => {
+      listener.close(() => {
+        resolve();
+      });
+    });
+    server.webhook.stop();
+    webhookServer.close();
+    // await db.close();
+  });
+
+  // afterEach(() => {
+  //   listener.close();
+  // });
+
+  it("should have a test server", async () => {
+    // webhookServer.use(bodyParser);
+    webhookServer.get("/self-test", (req, res) => {
+      res.end("self test was good");
+    });
+    const res = await fetch(`${testHost}/self-test`);
+    const text = await res.text();
+    expect(text).toEqual("self test was good");
+  });
+
+  it("should be able to receive the webhook event", async () => {
+    // create the webhook
+    let res = await client.post("/webhook", { ...mockWebhook });
+    let resJson = await res.json();
+    console.log("webhook body: ", resJson);
+    expect(res.status).toBe(201);
+    expect(resJson.blocking).toBe(true);
+    generatedWebhook = resJson;
+    res = await client.post("/webhook", {
+      ...mockWebhook,
+      name: "test 2",
+      blocking: false,
+    });
+    resJson = await res.json();
+    expect(resJson.blocking).toBe(false);
+    console.log("webhook body: ", resJson);
+    expect(res.status).toBe(201);
+    expect(resJson.name).toBe("test 2");
+    generatedWebhook2 = resJson;
+
+    client.jwtAuth = nonAdminToken["token"];
+    res = await client.post("/webhook", {
+      ...mockWebhook,
+      name: "test non admin",
+    });
+    resJson = await res.json();
+    console.log("webhook body: ", resJson);
+    expect(res.status).toBe(201);
+    expect(resJson.name).toBe("test non admin");
+    generatedWebhookNonAdmin = resJson;
+    client.jwtAuth = adminToken["token"];
+
+    // test endpoint
+    let resp;
+    webhookServer.use(bodyParser.json());
+    webhookServer.post("/webhook", (req, res) => {
+      console.log("WEBHOOK WORKS , body", req.body);
+      resp = 200;
+      res.end();
+    });
+
+    await server.db.queue.emit({
+      id: "webhook_test_12",
+      time: Date.now(),
+      channel: "test.channel",
+      event: "streamStarted",
+      streamId: resJson.id,
+      userId: nonAdminUser.id,
+      isConsumed: false,
+    });
+
+    await sleep(3000);
```

Comment:
I wonder if we could use a "semaphore-like" approach here instead of a sleep, but I think there's no built-in abstraction in node for that :/